presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorPageSourceProvider.java

```diff
@@ -25,5 +26,17 @@
     /**
      * @param columns columns that should show up in the output page, in this order
      */
-    ConnectorPageSource createPageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorSplit split, List<ColumnHandle> columns);
+    @Deprecated
```

Comment:
It feels like we should simply migrate it to avoid further confusions

@highker, @hellium01 Thoughts?